src/Sylius/Bundle/ApiBundle/Filter/TaxonFilter.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Filter;
+
+use ApiPlatform\Core\Api\IriConverterInterface;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Filter\AbstractContextAwareFilter;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use Doctrine\ORM\QueryBuilder;
+use Doctrine\Persistence\ManagerRegistry;
+use Sylius\Component\Taxonomy\Model\TaxonInterface;
+
+/**
+ * It should be replaced with generic collection filter after fixed: https://github.com/api-platform/api-platform/issues/1868
+ */
```

Comment:
now in this filter is more logic (we are getting child taxon) so this comment isn't true :)
```suggestion
```